plugins/artifacts/artifacts.go

```diff
@@ -186,6 +385,11 @@ func (tp *taskPlugin) processFile(result engines.ResultSet, a artifact) {
 		mtype = unknownMimetype
 	}
 
+	// Compute artifact hash for chain-of-trust
+	if err = tp.hashArtifact(a.Name, r); err != nil {
+		goto error
```

Comment:
I am not an anti-goto person, but if we invert the logic to `err == nil` and move the `UploadS3Artifact` call into the `if` clause, I think we can get rid of it.

Reply:
Nice, catch I used to have multiple things that had goto error, but now there is only one...